api/middleware/middleware.js

```diff
@@ -1,17 +1,65 @@
+const Users = require(`../users/users-model`)
+const Posts = require(`../posts/posts-model`)
+
+
 function logger(req, res, next) {
-  // DO YOUR MAGIC
+  console.log(`The request method is ${req.method}, the request url is ${req.url}, and the time is ${Date.now()}`)
+  next()
 }
 
-function validateUserId(req, res, next) {
-  // DO YOUR MAGIC
-}
+async function validateUserId(req, res, next) {
+  try{
+  const {id} = req.params
+  const user = await Users.getById(id)
+
+  if(user){
+    req.user = user
+    next()
+  }
+  else{
+    next({status: 404,
```

Comment:
Nit: Since you want to end the response here with 404 instead of sending it to next you can 
`return res.status(404).json({message: 'user not found'})` and it will stop the call and send the response here